main.go

```diff
@@ -164,10 +164,24 @@ func runServer(c *cli.Context) {
 	opts.OnConnectionLost = func(client mqtt.Client, err error) {
 		log.Printf("Error: Connection to %s lost: %s", c.String("endpoint"), err)
 		// update the "broker up" metric (down)
-		gaugeMetrics["up"].Set(0)
+		gaugeMetrics["broker_connection_up"].Set(0)
+		// try to reconnect
+		mqttConnect()
```

Comment:
Is this actually required? When connecting after having used [NewClientOptions](https://godoc.org/github.com/eclipse/paho.mqtt.golang#NewClientOptions), AutoReconnect is set to true.

I don't know for sure, but I think the code will reconnect by itself even without this code.

Reply:
I don't think it's the exporter's responsibility (or even prometheus's responsibility) to care if the broker is up. 

If we're not scraping, there's a problem.  If the scrape is old, there's a problem.  Don't add dimensions or complexity to it.

I think we fall back to just publishing `last_scrape_time`, and something else downstream will alert if it's too high (if that's prometheus's or alertmanager's job).